src/wallet/api/wallet.cpp

```diff
@@ -2463,6 +2464,61 @@ PendingTransaction* WalletImpl::stakePending(const std::string& sn_key_str, cons
   return transaction;
 }
 
+StakeUnlockResult* WalletImpl::canRequestStakeUnlock(const std::string &sn_key)
+{
+    tools::wallet2::request_stake_unlock_result res = {};
+
+    crypto::public_key snode_key;
+    if (!tools::hex_to_type(sn_key, snode_key))
+    {
+      res.success = false;
+      res.msg = "Failed to Parse Service Node Key";
+      StakeUnlockResultImpl stake_unlock_result(res);
+      return &stake_unlock_result;
+    }
+
+    StakeUnlockResultImpl stake_unlock_result(m_wallet->can_request_stake_unlock(snode_key));
+    return &stake_unlock_result;
+}
+
+StakeUnlockResult* WalletImpl::requestStakeUnlock(const std::string &sn_key)
+{
+    tools::wallet2::request_stake_unlock_result res = {};
+
+    crypto::public_key snode_key;
+    if (!tools::hex_to_type(sn_key, snode_key))
+    {
+      res.success = false;
+      res.msg = "Failed to Parse Service Node Key";
+      StakeUnlockResultImpl stake_unlock_result(res);
+      return &stake_unlock_result;
```

Comment:
As above regarding returning a pointer to local variable.  (And likewise for the other returns in this function).